gson/src/main/java/com/google/gson/DateFormatType.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gson;
+
+public enum DateFormatType {
+	/**
+	 * Use custom {@link java.text.DateFormat} to format Date at serialization if
+	 * was set. Otherwise is same as {@code DateFormatType.DEFAULT}
+	 */
+	CUSTOM,
```

Comment:
Add few sample DateTypeAdapters was my first idea, but then I thought that maybe we would like the adapter to be able to parse multiple date formats, and also be able to specify the output date format. That's how I came up with an idea to create DateFormatters and simply specify to the adapter which of them to use. I agree that too much customization adds more complexity.

Maybe we could provide an EnumSet of DateFormatTypes for parsing, and one Enum DateFormatType for output. ex: 
`new DefaultDateTypeAdapter(EnumSet.of(ISO8601, MILLIS, EN_US), DateTypeFormat.MILLIS);`
That way the adapter will use ISO8601, MILLIS and EN_US formatters to parse, and MILLIS for output.

I can remove the CUSTOM and DEFAULT format types, so only standard date formats can be used for output.

In the case nothing was specified, the adapter will use the defaults: EN_US, LOCAL and ISO8601 to parse and EN_US for output, as it does now.

What do you think?